src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -439,47 +439,50 @@ static internal bool IsNonAttributedTypeValidForSerialization(Type type)
-                "System.Tuple`8",
+        private static readonly Dictionary<string, string[]> s_knownSerializableTypeInfos = new Dictionary<string, string[]> {
+            { "System.Collections.Generic.KeyValuePair`2", Array.Empty<string>() },
+            { "System.Collections.Generic.Queue`1", new [] { "_syncRoot" } },
+            { "System.Collections.Generic.Stack`1", new [] {"_syncRoot" } },
+            { "System.Collections.ObjectModel.ReadOnlyCollection`1", new [] {"_syncRoot" } },
+            { "System.Collections.ObjectModel.ReadOnlyDictionary`2", new [] {"_syncRoot", "_keys","_values" } },
+            { "System.Tuple`1", Array.Empty<string>() },
+            { "System.Tuple`2", Array.Empty<string>() },
+            { "System.Tuple`3", Array.Empty<string>() },
+            { "System.Tuple`4", Array.Empty<string>() },
+            { "System.Tuple`5", Array.Empty<string>() },
+            { "System.Tuple`6", Array.Empty<string>() },
+            { "System.Tuple`7", Array.Empty<string>() },
+            { "System.Tuple`8", Array.Empty<string>() },
+            { "System.Collections.Queue", new [] {"_syncRoot" } },
+            { "System.Collections.Stack", new [] {"_syncRoot" } },
+            { "System.Globalization.CultureInfo", Array.Empty<string>() },
+            { "System.Version", Array.Empty<string>() },
         };
 
+        private static string GetGeneralTypeName(Type type)
+        {
+            TypeInfo typeInfo = type.GetTypeInfo();
+            return type.GetTypeInfo().IsGenericType && !typeInfo.IsGenericParameter
+                ? typeInfo.GetGenericTypeDefinition().FullName
+                : type.FullName;
+        }
+
         internal static bool IsKnownSerializableType(Type type)
         {
             // Applies to known types that DCS understands how to serialize/deserialize
             //
+            string typeFullName = GetGeneralTypeName(type);
 
-            // Ajdust for generic type
-            if (type.GetTypeInfo().IsGenericType && !type.GetTypeInfo().IsGenericTypeDefinition)
-            {
-                type = type.GetGenericTypeDefinition();
-            }
+            return s_knownSerializableTypeInfos.ContainsKey(typeFullName)
+                || Globals.TypeOfException.IsAssignableFrom(type);
+        }
 
-            // Check for known types
-            if (Enumerable.Contains(s_knownSerializableTypeNames, type.FullName))
-            {
-                return true;
-            }
-            //Enable ClassDataContract to give support to Exceptions.
-            if (Globals.TypeOfException.IsAssignableFrom(type))
-                return true;
+        internal static bool IsNonSerializedMember(Type type, string memberName)
+        {
+            string typeFullName = GetGeneralTypeName(type);
 
-            return false;
+            return s_knownSerializableTypeInfos.ContainsKey(typeFullName)
+                && s_knownSerializableTypeInfos[typeFullName].Contains(memberName);
```

Comment:
Also just noticed: Instead of doing the key lookup twice, it can be done once:

```csharp
string[] members;
return s_knownSerializableTypeInfos.TryGetValue(typeFullName, out members)
    && members.Contains(memberName);
```